contrib/mesos/pkg/executor/service/service.go

```diff
@@ -288,8 +381,20 @@ func (ks *KubeletExecutorServer) createAndInitKubelet(
 		MaxContainers:      kc.MaxContainerCount,
 	}
 
+	// create main pod source
 	pc := kconfig.NewPodConfig(kconfig.PodConfigNotificationIncremental, kc.Recorder)
 	updates := pc.Channel(MESOS_CFG_SOURCE)
+	go func() {
+		for u := range execUpdates {
```

Comment:
but KubeletExecutor.done is used everywhere in executor.go. Passing it through the whole control flow is not much nicer.